pipeline/load.py

```diff
@@ -0,0 +1,288 @@
+    Returns -1 if it cannot be found.
+    """
+    search_query = f"SELECT country_id FROM country WHERE country_name = '{
+        country_name}';"
+    cursor.execute(search_query)
+    country_id = cursor.fetchone()
+    if not country_id:
+        logging.info("Unable to find id for %s", country_name)
+        return -1
+    return country_id[0]
+
+
+def get_id_from_artist(artist_name: str, cursor: psycopg2.extensions.cursor) -> int:
+    """
+    Retrieves the artist id of a 
+    given artist from the database.
+    Returns -1 if it cannot be found.
+    """
+    search_query = "SELECT artist_id FROM artist WHERE artist_name = %s;"
+    cursor.execute(search_query, (artist_name,))
+    artist_id = cursor.fetchone()
+    if not artist_id:
+        logging.info("Unable to find id for %s", artist_name)
+        return -1
+    return artist_id[0]
+
+
+def get_id_from_release_type(release_type: str, cursor: psycopg2.extensions.cursor) -> int:
+    """
+    Retrieves the release type id of a 
+    given release type.
+    Returns -1 if it cannot be found.
+    """
+    search_query = "SELECT type_id FROM type WHERE type_name = %s;"
+    cursor.execute(search_query, (release_type.lower(),))
+    type_id = cursor.fetchone()
+    if not type_id:
+        logging.info("Unable to find id for %s", release_type)
+        return -1
+    return type_id[0]
+
+
+def insert_country(country_name: str, cursor: psycopg2.extensions.cursor) -> None:
+    """
+    Inserts a country into the database if it doesn't already exist.
+    Logs whether the country was added or already exists.
+    """
+    try:
+        check_query = "SELECT EXISTS (SELECT 1 FROM country WHERE country_name = %s);"
+        cursor.execute(check_query, (country_name,))
+        exists = cursor.fetchone()[0]
+
+        if exists:
+            logging.info(
+                "Country '%s' already exists in the database.", country_name)
+        else:
+            insert_query = "INSERT INTO country (country_name) VALUES (%s);"
+            cursor.execute(insert_query, (country_name,))
+            logging.info(
+                "Country '%s' was added to the database.", country_name)
+    except psycopg2.Error:
+        logging.error("Error inserting country: '%s'", country_name)
+
+
+def insert_artist(artist_name: str, country: str, cursor: psycopg2.extensions.cursor) -> None:
+    """
+    Inserts an artist into the database if they doesn't already exist.
+    Logs whether the artist was added or already exists.
+    """
+    country_id = get_id_from_country(country, cursor)
+    try:
+        check_query = """SELECT artist_id FROM artist
+                        WHERE (artist_name = %s
+                        AND country_id = %s);"""
+        cursor.execute(check_query, (artist_name, country_id))
+        artist_id = cursor.fetchone()
+
+        if artist_id:
+            logging.info(
+                "Artist '%s' already exists in with ID: %s.", artist_name, artist_id[0])
```

Comment:
Maybe the same goes for all of our insert functions as with countries when it comes to logging?

Reply:
I think countries makes sense as to not log when it exists, but the number of artists are not really limited by anything, so I think having this log here is fine.